frame/staking/src/pallet/mod.rs

```diff
@@ -318,8 +318,10 @@ pub mod pallet {
 	pub type MinCommission<T: Config> = StorageValue<_, Perbill, ValueQuery>;
 
 	/// Map from all (unlocked) "controller" accounts to the info regarding the staking.
+	///
+	/// Note: All the reads and mutations to this storage *MUST* be done through the methods exposed
+	/// by [`StakingLedger`] to ensure data and lock consistency.
 	#[pallet::storage]
```

Comment:
we could actually put this into a private mod now and use `frame::use_parts()` or something like that? not that it is super critical here, but I'd like to see pallet splitting in action, if it is even useful @gupnik might help here.